### PR TITLE
Add zxing-cpp formula

### PR DESCRIPTION
**Pull request by mjbshaw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Adds the ZXing C++ port to Homebrew. The C++ port was deprecated and removed from the main ZXing project after v2.3.0, and its development has moved to a new repo ([see details here](https://github.com/zxing/zxing#zxing-based-third-party-open-source-projects)).

### CODE REVIEW

File: Formula/zxing-cpp.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  def install
+    if build.stable?
+      cd "cpp"
+    end
+
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args
+      system "make"
+
+      if build.stable?
```
Comment (fxcoudert) [inline Formula/zxing-cpp.rb]: Isn't there a `make install` for stable?

File: Formula/zxing-cpp.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        lib.install "libzxing.a"
```
Comment (fxcoudert) [inline Formula/zxing-cpp.rb]: Can we build a shared library instead of static?

File: Formula/zxing-cpp.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      end
+
+      if build.head?
+        system "make", "install"
+      end
+    end
+  end
+
+  test do
+    system bin/"zxing"
```
Comment (fxcoudert) [inline Formula/zxing-cpp.rb]: Please include a test that actually does something. Consider installing a test image for that.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Review by mjbshaw — COMMENTED** [review]
I added a proper test.

File: Formula/zxing-cpp.rb (hunk 3 above)
Comment (mjbshaw) [inline Formula/zxing-cpp.rb]: Done.

File: Formula/zxing-cpp.rb (hunk 1 above)
Comment (kriansa) [inline Formula/zxing-cpp.rb]: @mjbshaw I believe you made a typo on **install** (not _intall_)

Comment (mjbshaw) [inline Formula/zxing-cpp.rb]: Oops, that command had a typo. But it really doesn't have `make install`:

```
$ make install
make: *** No rule to make target `install'.  Stop.
$ grep install Makefile
# Prepare targets for installation.
preinstall: all
	$(MAKE) -f CMakeFiles/Makefile2 preinstall
.PHONY : preinstall
# Prepare targets for installation.
preinstall/fast:
	$(MAKE) -f CMakeFiles/Makefile2 preinstall
.PHONY : preinstall/fast
```

Comment (dunn) [inline Formula/zxing-cpp.rb]: Could you ask the developers if they'll add one?

Comment (mjbshaw) [inline Formula/zxing-cpp.rb]: I don't think that would work, given that [the C++ port was deleted in 2013](https://github.com/zxing/zxing/commit/4856e86a46a5836d7ea793a78bea5637c85fe7c6#diff-ac731f2fedd74089d1149d83620eeb92) from the main repo. [zxing-cpp](https://github.com/glassechidna/zxing-cpp) forked that code, and is the unofficial-official active C++ zxing repo (but with no releases, currently).

I know it's a sorry state of affairs, but it's one that is common to barcode/QR decoding libraries, and unfortunately what C++ developers have to work with for the time being.

**Comment by fxcoudert** [comment]
Thanks @mjbshaw for your contribution to Homebrew!

**Pull request closed without merging**